config/config.go

```diff
@@ -136,19 +138,30 @@ const (
 )
 
 type Metrics struct {
-	Type   MetricsType `mapstructure:"type"`
-	Influx Influx      `mapstructure:"influx"`
+	Type       MetricsType       `mapstructure:"type"`
+	Influx     InfluxMetrics     `mapstructure:"influx"`
+	Prometheus PrometheusMetrics `mapstructure:"prometheus"`
 }
 
 func (cfg *Metrics) validateAndLog() {
-	log.Infof("config.metrics.type: %s", cfg.Type)
-	switch cfg.Type {
-	case MetricsNone:
-	case MetricsInflux:
-		cfg.Influx.validateAndLog()
-	default:
-		log.Fatalf(`invalid config.metrics.type: %s. It must be "none" or "influx"`, cfg.Type)
+	var err error
+	err = cfg.checkMetricsEnabled()
+
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+}
+
+func (cfg *Metrics) checkMetricsEnabled() error {
+	var err error
+	cfg.Influx.Enabled = (cfg.Type == MetricsInflux || cfg.Influx.Enabled)
+	cfg.Influx.Enabled = cfg.Influx.validateAndLogMetricsData()
+	cfg.Prometheus.Enabled = cfg.Prometheus.Enabled && cfg.Prometheus.validateAndLogMetricsData()
+
+	if !cfg.Influx.Enabled && !cfg.Prometheus.Enabled && cfg.Type != MetricsNone {
+		err = fmt.Errorf("No metrics correctly specified in configuration file")
```

Comment:
This should be changed to a log warning. I don't consider it an error to run PBC without any metrics. Since we account for `cfg.Type` in `cfg.Influx.Enabled`, this can just be `!cfg.Influx.Enabled && !cfg.Prometheus.Enabled`.

Reply:
This logic in line 161 was already in place before this PR and I believe the reason behind it is: if a metrics system other than influx or prometheus was specified, execution stops with an error so we don't expect to get metrics that won't be there. The fact that `cfg.Type != MetricsNone` evaluate to true, opens up the possibility for `cfg.Type` to equal `"Trendalyze"`, `"Amazon Timestream"`, or any other metric system that we don't support.  I prefer the system to use the `cfg.Influx.Enabled` and `cfg.Prometheus.Enabled` flags instead of using the `cfg.Type` field but we decided to let it there for backwards compatibility.
```
155 func (cfg *Metrics) checkMetricsEnabled() error {
156     var err error
157 +--  3 lines: cfg.Influx.Enabled = (cfg.Type == MetricsInflux || cfg.Influx.Enabled)--
160
161     if !cfg.Influx.Enabled && !cfg.Prometheus.Enabled && cfg.Type != MetricsNone {
162         err = fmt.Errorf("No metrics correctly specified in configuration file")
163     }
164     return err
165 }
config/config.go
```
I can still change it to a warning. What do you think?

Reply:
I believe the new logic illustrates this better. Modified.